zebrad/tests/acceptance.rs

```diff
@@ -784,7 +826,7 @@ fn create_cached_database_height(network: Network, height: Height) -> Result<()>
 
     let dir = PathBuf::from("/zebrad-cache");
     let mut child = dir
-        .with_config(config)?
+        .with_exact_config(&config)?
```

Comment:
@dconnolly @yaahc this is the actual fix, all the rest is refactoring and type system enforcement